### PR TITLE
Add some e2e tests for TTL for entity handles.

### PR DESCRIPTION
**Pull request by galganif**

### CODE REVIEW

File: javatests/arcs/android/entity/TtlHandleTest.kt [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    @Before
+    fun setUp() {
+        databaseManager = AndroidSqliteDatabaseManager(ApplicationProvider.getApplicationContext())
+        DriverAndKeyConfigurator.configure(databaseManager)
+        SchemaRegistry.register(DummyEntity)
+    }
+
+    @Test
+    fun singletonWithExpiredEntities() = runBlockingTest {
+        val time = FakeTime()
```
Comment (jblebrun) [inline javatests/arcs/android/entity/TtlHandleTest.kt:51]: Can this be created as a property of the test class, as part of `setUp`? Then the helper methods can just use the time field directly, and there's no need to pass it to the handle creation helpers.

File: javatests/arcs/android/entity/TtlHandleTest.kt [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        handle1.store(entity1)
+        handle1.store(entity2)
+        
+        // Entities in handle2 are not expired.
+        time.millis = System.currentTimeMillis()
+        handle2.store(entity1)
+        handle2.store(entity3)
+        handle2.store(entity4)
+        
+        time.millis = 1L
```
Comment (jblebrun) [inline javatests/arcs/android/entity/TtlHandleTest.kt:164]: time moving in both directions feels a bit odd...

File: javatests/arcs/android/entity/TtlHandleTest.kt [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            num = 1.0
+            texts = setOf("1", "one")
+        }
+        // Set a time in the past. So that this entity is already expired.
+        time.millis = 1L
+        handle.store(entity1)
+        // Then set time to now.
+        time.millis = System.currentTimeMillis()
+
+        // TODO: once expired entities are filtered out on read, this should be null.
```
Comment (jblebrun) [inline javatests/arcs/android/entity/TtlHandleTest.kt:63]: Is there a `b/` that can be referenced here?

File: javatests/arcs/android/entity/TtlHandleTest.kt [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        assertThat(handle.fetch()).isEqualTo(entity1)
+        
+        // Simulate periodic job triggering.
+        databaseManager.removeExpiredEntities()
+
+        assertThat(handle.fetch()).isEqualTo(null)
+
+        // Creatie a new handle manager with a new storage proxy to confirm entity1 is gone and the
```
Comment (jblebrun) [inline javatests/arcs/android/entity/TtlHandleTest.kt:71]: nit: typo on `Create`

File: javatests/arcs/android/entity/TtlHandleTest.kt [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+        assertThat(handle.fetchAll()).containsExactly(entity2)
+
+        val entity3 = DummyEntity().apply {
+            num = 3.0
+            texts = setOf("3", "three")
+        }
+        handle.store(entity3)
+        assertThat(handle.fetchAll()).containsExactly(entity2, entity3)
+
+        // Creatie a new handle manager with a new storage proxy to confirm entity1 is gone and the
```
Comment (jblebrun) [inline javatests/arcs/android/entity/TtlHandleTest.kt:122]: same typo

File: javatests/arcs/android/entity/TtlHandleTest.kt (hunk 2 above)
Comment (galganif) [reply javatests/arcs/android/entity/TtlHandleTest.kt:164]: Good point, I have reworked the test to insert all entities at the same time

File: javatests/arcs/android/entity/TtlHandleTest.kt (hunk 1 above)
Comment (galganif) [reply javatests/arcs/android/entity/TtlHandleTest.kt]: Done

File: javatests/arcs/android/entity/TtlHandleTest.kt (hunk 3 above)
Comment (galganif) [reply javatests/arcs/android/entity/TtlHandleTest.kt]: Done

File: javatests/arcs/android/entity/TtlHandleTest.kt (hunk 4 above)
Comment (galganif) [reply javatests/arcs/android/entity/TtlHandleTest.kt]: Done

File: javatests/arcs/android/entity/TtlHandleTest.kt (hunk 5 above)
Comment (galganif) [reply javatests/arcs/android/entity/TtlHandleTest.kt]: Done

**Comment by galganif** [comment]
> Looks great to me. I do wonder if we could eventually move these tests into `HandleManagerTestBase` so that they can run in different handle manager configurations?

Not sure, I specifically wanted to test this combination of handles/storage proxies, and here I am not testing handles, I am mostly testing the removeExpiredEntities method works as intended. There are already other TTL tests in HandleManagerTestBase for handle specific logic.

**Comment by galganif** [comment]
tap test: cl/305571112

**Pull request merged**